File: Samples/State Pattern/Sample_0 - CPP/IdleState.cpp
#include "IdleState.h"

void IdleState::Enter()
{
	cout << "IdleState called Enter!" << endl;
}

void IdleState::Execute()
{
	cout << "IdleState called Execute!" << endl;
}

void IdleState::Exit()
{
	cout << "IdleState called Exit!" << endl;
}

File: Samples/State Pattern/Sample_0 - CS/IdleState.cs
namespace Sample
{
    internal class IdleState : State
    {
        public IdleState()
        {
            _name = "IdleState";
        }

        public override void Enter()
        {
            Console.WriteLine("IdleState called Enter");
        }

        public override void Execute()
        {
            Console.WriteLine("IdleState called Execute");
        }

        public override void Exit()
        {
            Console.WriteLine("IdleState called Exit");
        }
    }
}
